LSP hover of typescript-tools and tailwindcss conflicting

I have typescript-tools and tailwindcss LSPs installed and attached to `.jsx` and `.tsx` files.

When trying to hover on a tailwindcss class in a JSX fragment, there is no hover window that appears.

However, when I run `:LspStop typescript-tools`, and ensure that the tailwindcss LSP is the only server attached, the hover window appears as expected.

I have also tried setting up ts_ls without typescript-tools and the hover works as expected.

Yes, the PR does resolve the issue of the hovers. However, I am running into a new issue with the changes.

I am currently using the following plugin for signature help (https://github.com/ray-x/lsp_signature.nvim) and attaching it on the `LspAttach` autocommand. The minimal `LspAttach` autocommand is as follows:

```lua
vim.api.nvim_create_autocmd("LspAttach", {
    group = augroup,
    callback = function(args)
        local bufnr = args.buf

        require("lsp_signature").on_attach({
            floating_window = false,
            hint_enable = true,
            hint_prefix = "",
            hint_scheme = "Comment",
        }, bufnr)
    end,
})
```

When opening a Typescript file, typescript-tools attach, but the LSP methods are not working (goto definition, hover).

In the `lsp.log` file, the relevant logs are as follows:

```
[ERROR] ...ols/tsserver.lua:104	"tsserver"	"Unexpected error while handling response: "	"textDocument/signatureHelp"
[ERROR] ...ols/tsserver.lua:105	"tsserver"	"vim/shared.lua:0: t: expected table, got nil\nstack traceback:\n\t[C]: in function 'error'\n\tvim/shared.lua: in function 'validate'\n\tvim/shared.lua: in function 'make_signatures'\n\t...pescript-tools/protocol/text_document/signature_help.lua:107: in function <...pescript-tools/protocol/text_document/signature_help.lua:89>"
```

The log does come from the new log statement added in the PR.

I did not have the time to find the bug with the signature help plugin, so I am not sure if it is related to typescript-tools, but its just my observations.

--- a/lua/typescript-tools/protocol/text_document/document_highlight.lua
+++ b/lua/typescript-tools/protocol/text_document/document_highlight.lua
@@ -30,6 +30,7 @@ function M.handler(request, response, params)
   -- https://github.com/microsoft/TypeScript/blob/8a1b85880f89c9cff606c5844e8883e5f483c7db/lib/protocol.d.ts#L844
   if not body[1] then
     response(nil)
+    return
   end
 
   response(vim.tbl_map(function(item)

--- a/lua/typescript-tools/protocol/text_document/document_symbol.lua
+++ b/lua/typescript-tools/protocol/text_document/document_symbol.lua
@@ -40,8 +40,9 @@ function M.handler(request, response, params)
 
   -- tsserver protocol reference:
   -- https://github.com/microsoft/TypeScript/blob/8a1b85880f89c9cff606c5844e8883e5f483c7db/lib/protocol.d.ts#L2561
-  if #body.childItems == 0 then
+  if not body.childItems or #body.childItems == 0 then
     response(nil)
+    return
   end
 
   response(vim.tbl_map(map_document_symbol, remove_aliases(body.childItems)))

--- a/lua/typescript-tools/protocol/text_document/hover.lua
+++ b/lua/typescript-tools/protocol/text_document/hover.lua
@@ -17,6 +17,11 @@ function M.handler(request, response, params)
 
   local body = coroutine.yield()
 
+  if body.success == false then
+    response(nil)
+    return
+  end
+
   local contents = {
     utils.tsserver_docs_to_plain_text(body.documentation),
     utils.tsserver_make_tags(body.tags),

--- a/lua/typescript-tools/protocol/text_document/signature_help.lua
+++ b/lua/typescript-tools/protocol/text_document/signature_help.lua
@@ -101,6 +101,11 @@ function M.handler(request, response, params)
 
   local body = coroutine.yield()
 
+  if body.success == false then
+    response(nil)
+    return
+  end
+
   -- tsserver protocol reference:
   -- https://github.com/microsoft/TypeScript/blob/96894db6cb5b7af6857b4d0c7f70f7d8ac782d51/lib/protocol.d.ts#L1980
   response {

--- a/lua/typescript-tools/tsserver.lua
+++ b/lua/typescript-tools/tsserver.lua
@@ -94,7 +94,19 @@ function Tsserver:handle_response(response)
 
   local handler = metadata.handler
 
-  coroutine.resume(handler, response.body or response, response.command or response.event)
+  -- NOTE: for some reason, progress tries to resume a dead coroutine
+  if coroutine.status(handler) ~= "dead" then
+    local _, err =
+      coroutine.resume(handler, response.body or response, response.command or response.event)
+
+    if err then
+      local _ = log.error()
+        and log.error("tsserver", "Unexpected error while handling response: ", metadata.method)
+      local _ = log.error() and log.error("tsserver", debug.traceback(handler, err))
+
+      return false, seq
+    end
+  end
 
   self.pending_requests[seq] = nil
   self.requests_metadata[seq] = nil

